paradox-nakadi-consumer-core/src/test/java/de/zalando/paradox/nakadi/consumer/core/client/impl/ClientImplTest.java

```diff
@@ -47,61 +70,105 @@ public void testShouldThrowWhenRequestingUnconsumedEventsForDifferentEventTypes(
+    public void testGetNakadiPartitionsFound() throws IOException {
+        final MockWebServer mockNakadi = new MockWebServer();
+        mockNakadi.enqueue(new MockResponse().setBody(NAKADI_RESPONSE_PARTITIONS));
+        mockNakadi.start(NAKADI_PORT);
+
         final List<NakadiPartition> list = clientImpl.getPartitions(ORDER_RECEIVED).toBlocking().value();
         assertThat(list).isNotEmpty();
+
+        mockNakadi.close();
     }
 
     @Test
-    @Ignore("Needs Nakadi server")
-    public void testGetNakadiPartitionsNotFound() {
+    public void testGetNakadiPartitionsNotFound() throws IOException {
+        final MockWebServer mockNakadi = new MockWebServer();
+        mockNakadi.enqueue(new MockResponse().setResponseCode(404).setBody(NAKADI_RESPONSE_NOT_FOUND));
+        mockNakadi.start(NAKADI_PORT);
+
         final List<NakadiPartition> list = clientImpl.getPartitions(EventType.of("__not_configured__"))
                                                      .onErrorReturn(throwable -> Collections.emptyList()).toBlocking()
                                                      .value();
         assertThat(list).isEmpty();
+
+        mockNakadi.close();
     }
 
     @Test
-    @Ignore("Needs Nakadi server")
-    public void testGetNakadiEventPayload() {
+    public void testGetNakadiEventPayload() throws IOException {
+        final MockWebServer mockNakadi = new MockWebServer();
+        mockNakadi.enqueue(new MockResponse().setBody(NAKADI_RESPONSE_ONE_ORDER));
+        mockNakadi.start(NAKADI_PORT);
+
         final EventTypeCursor cursor = EventTypeCursor.of(EventTypePartition.of(ORDER_RECEIVED, "0"), "0");
         final String event = clientImpl.getEvent(cursor).toBlocking().value();
         assertThat(event).startsWith("{\"metadata\":{");
         assertThat(event).contains("\"order_number\"");
+
+        mockNakadi.close();
     }
 
     @Test
-    @Ignore("Needs Nakadi server")
-    public void testGetNakadiEventStream() {
+    public void testGetNakadiEventStream() throws IOException {
+        final MockWebServer mockNakadi = new MockWebServer();
+        mockNakadi.enqueue(new MockResponse().setBody(NAKADI_RESPONSE_ONE_ORDER));
+        mockNakadi.start(NAKADI_PORT);
+
         final EventTypeCursor cursor = EventTypeCursor.of(EventTypePartition.of(ORDER_RECEIVED, "0"), "BEGIN");
         final String stream = clientImpl.getContent(cursor).toBlocking().value();
-        assertThat(stream).startsWith("{\"cursor\":{\"partition\":\"0\",\"offset\":\"0\"}");
+        assertThat(stream).startsWith("{\"cursor\":{\"partition\":\"0\",\"offset\":\"000000000000000000\"}");
         assertThat(stream).contains("\"order_number\"");
+
+        mockNakadi.close();
     }
 
     @Test
-    @Ignore("Needs Nakadi server")
-    public void testGetNakadiEventObject() {
+    public void testGetNakadiEventObject() throws IOException {
+        final MockWebServer mockNakadi = new MockWebServer();
```

Comment:
I would move the creation of the mock server to `setUp()` and `tearDown()`:

```
    @Before
    public void setUp() throws Exception {
        mockNakadi = new MockWebServer();
        mockNakadi.start(NAKADI_PORT);
    }

    @After
    public void tearDown() throws Exception {
        mockNakadi.close();
    }
```

This way, even in case of an exception, the server is closed correctly.